src/reynir_correct/pattern.py

```diff
@@ -1440,34 +1465,35 @@ def create_patterns(cls) -> None:
                     None,
                 )
             )
-
-            # Catch "Jón leggur hann af velli."
+            # Catch "Það hefur ekki markað upphafið af því."
             p.append(
                 (
-                    "leggja",  # Trigger lemma for this pattern
-                    "VP > { VP > { VP > { 'leggja' } } PP > { P > { 'af' } NP > { 'völlur' } } }",
-                    cls.wrong_preposition_leggja_af,
+                    "upphefja",  # Trigger lemma for this pattern
+                    "VP > { VP > { NP > { 'markaður' } VP > { 'upphefja' } } PP > { 'af' } }",
+                    cls.wrong_preposition_marka_af,
                     None,
                 )
             )
-            # Catch "Jón hefur lagt hann af velli."
+
+            # Catch "Jón leggur hann (ekki) af velli."
             p.append(
                 (
                     "leggja",  # Trigger lemma for this pattern
-                    "VP > { VP > { VP > { VP > { 'leggja' } } } PP > { P > 'af' NP > 'völlur' } }",
+                    "VP > { VP > { VP > { 'leggja' } } PP > { P > { 'af' } NP > { ( 'völlur'|'vell' ) } } }",
```

Comment:
'vell' would be a wrong parse. And 'völlur' is probably better as a trigger lemma than "leggja", as it is much less common.

Reply:
Done!